### ISSUE TITLE
Train focusing on 100 first instances with higher scores

### ISSUE THREAD

**Issue by alejandrogomez97** [pycaret/pycaret#3694]
### Describe the feature you want to add to this project

New metric to focus only on instances with high scores

### Describe your proposed solution

Hi. I have a binary variable with two possible values, 0 or 1. It's an imbalanced problem, 98% of my data is 0 and 2% is 1. My objective is to give a list of 100 the instances of the test set which have more probabilities to be 1. 

Until now, I was facing this with a binary lightgbm classifier using the average precision metric. The problem is that using this metric i'm focusing on predicting all the data and i'm only interested in predicting correctly the 100 instances with higher scores. Intuitively, if i used a metric to train that depended on the predicted score of the algorithm I could have less false positives in the first 100 instances. How can I do so? Is there any metric that can help me solve this issue in pycaret? This is a very common issue and it would be nice to have one.

**Comment by moezali1** [comment]
@alejandrogomez97 The top N predictions are only known and can be achieved by sorting the test set. But for this all the test set has to be inferred. I am not sure if there is any way to select Top N before inferring the entire test set. Did I understand it wrong? please let me know.

**Comment by alejandrogomez97** [comment]
> @alejandrogomez97 The top N predictions are only known and can be achieved by sorting the test set. But for this all the test set has to be inferred. I am not sure if there is any way to select Top N before inferring the entire test set. Did I understand it wrong? please let me know.

**Comment by alejandrogomez97** [comment]
It's ok to infer the whole test set. The thing is that i need to construct my algorithm optimizing a metric that gives more importance to errors in the top 100 scores. Finally i changed the default metric i was using for a customized metric which is the mean real value of the 100 highest scores observations in the test set. 

Using the average precision metric, the algorithm gives equal importance to observations that are probably ones than to observations that are probably zeros. That being said, it will try to improve precision in the whole data equally, so it will sacrifice accuracy in top 100 high score observations to improve accuracy in the remainig 99% of the data. 

As in my case
